data.tf

```
@@ -1,7 +1,8 @@
 locals {
+  api_gw_enabled = local.is_govcloud || var.api_gw_enabled == true ? toset(["enabled"]) : []
   is_govcloud     = data.aws_partition.current.partition == "aws-us-gov"
   each_govcloud   = local.is_govcloud ? toset(["enabled"]) : []
-  each_commercial = local.is_govcloud ? [] : toset(["enabled"])
+  each_commercial = local.is_govcloud || var.api_gw_enabled == true ? [] : toset(["enabled"])
```

Maybe change to `function_url_enabled`?

```suggestion
  function_url_enabled = local.is_govcloud || var.api_gw_enabled ? [] : toset(["enabled"])
```